ui/pages/confirmations/hooks/useAssetDetails.js

```diff
@@ -65,6 +69,10 @@ export function useAssetDetails(tokenAddress, userAddress, transactionData) {
     prevTokenBalance,
   ]);
 
+  if (!tokenAddress && !userAddress && !transactionData) {
+    return {};
+  }
```

Comment:
Maybe we can avoid above condition as `currentAsset` will be set only if above condition is not true.

Reply:
This is needed because the hook is called on title.tsx even when currentConfirmation is a signature.

Reply:
Right, but along similar lines, if this block is moved to right before the above `useEffect` call, couldn't the same check on `:36-39` be removed? As `tokenAddress && !userAddress && !transactionData` can not change after instantiation, it seems safe to just skip the `useEffect` when they are all falsy?

Reply:
It's not clear to me that these variables cannot change after instantiation. In fact they are all in the dependency array of the useEffect.

Apart from that, it's generally preferred to avoid conditionally rendering hooks.

Reply:
> It's not clear to me that these variables cannot change after instantiation. I

Where could they? They only live within the scope of the body of the function `useAssetDetails`, and as such can not be modified elsewhere.

I also do not see them mutated anywhere withing the function?

Reply:
Since they are arguments of this hook, they could presumably be changed upstream, triggering the useEffect to rerun